### PR TITLE
Global styles: remove block gap control

### PR DESCRIPTION
**Pull request by ramonjd**
## What?

Remove the block gap controls from Global Styles sidebar. Props to @andrewserong for [noticing](https://github.com/WordPress/gutenberg/pull/37736#pullrequestreview-915188365).

## Why?

1. It's not supported at the root level in [ROOT_BLOCK_SUPPORTS](https://github.com/WordPress/gutenberg/blob/e2f00ff5ed1f4042386a48eedf4eafef36c9eb29/packages/edit-site/src/components/global-styles/hooks.js#L156)
2. It doesn't work at the individual block level in global styles anyway.

## How?
BANISHED!

## Testing Instructions
Before checking out this branch, ensure that blockGap doesn't:

- Appear at the root global styles level
- Work for block Global Style, e.g., edit the Group/Navigation/Social Icons (or any block that has blockGap support) block gap values.

Publish a page after you've set a few Global Style block gap values. Check the source of the published page for any trace of your `gap` value. Any?

1. Check out this branch `git checkout -t origin/remove/global-styles-block-gap-control`
2. Using a block theme, edit your site.
3. Check that Global Style block gap dimensions controls do not show at both the root level and for blocks that has blockGap support

### Before
Screen Shot 2022-03-21 at 3 28 09 pm

### After

Screen Shot 2022-03-21 at 3 40 19 pm
Screen Shot 2022-03-21 at 3 40 15 pm
Screen Shot 2022-03-21 at 3 40 05 pm

### CODE REVIEW

**Review by aaronrobertshaw — APPROVED** [review]
This tests as advertised for me. 👍 

✅ Confirmed on trunk that `blockGap` isn't displayed at the root global styles level
✅ Confirmed on trunk that `blockGap` global styles for individual blocks didn't work e.g. Social Icons
✅ Didn't find any reference to gap on published page
✅ After checking out this branch `blockGap` controls did not show at root or individual block levels

The changes LGTM. I may be missing a little context on the reasoning behind this PR so might pay to wait for @andrewserong's approval.

**Review by andrewserong — APPROVED** [review]
This is testing well for me, too, thanks for putting it together so quickly @ramonjd! 🤞 we'll be able to re-introduce the feature before too long 😀

| Block spacing is no longer available in global styles | Block spacing is still available within the post and site editors, just at the individual block level |
| --- | --- |
| image | image |

LGTM! 🚀

**Comment by youknowriad** [comment]
Hey folks, while removing this from the "blocks" global styles is a good call here, I'm a bit confused about why we need to remove it from the global level. I see that you mention that it didn't work, can we fix that instead?

I know that when I removed the CSS variables from the blocks block gap behavior, I intentionally kept the variable as a fallback in the generated styles to be able to support defining the block gap at the root level.

**Comment by mtias** [comment]
@ramonjd @andrewserong @aaronrobertshaw this was a very intentional and important feature of global styles to help design visual rhythm :D 

image

**Comment by ramonjd** [comment]
Thanks for the heads up @youknowriad and @mtias 

Revert commit here: https://github.com/WordPress/gutenberg/pull/39845

Was a bit of a shotgun approach admittedly. 

I'll look more closely at what is not working and create issues/fix it.

Related:
- https://github.com/WordPress/gutenberg/issues/39789

**Comment by mtias** [comment]
Thanks!

**Comment by andrewserong** [comment]
> I see that you mention that it didn't work, can we fix that instead?

Thanks for following up, @ramonjd! Yes, just confirming that the intention is definitely to fix it and have it working properly in time for WP 6.0.

**Pull request merged**
